src/lsm/lsm_cursor_bulk.c

```diff
@@ -31,7 +34,16 @@ __clsm_close_bulk(WT_CURSOR *cursor)
 	clsm->nchunks = 0;
 
 	/* Set ondisk, and flush the metadata */
-	F_SET(lsm_tree->chunk[0], WT_LSM_CHUNK_ONDISK);
+	F_SET(chunk, WT_LSM_CHUNK_ONDISK);
+	/*
+	 * Setup a generation in our chunk, so that future LSM merges choose
+	 * reasonable sets of chunks.
+	 */
+	avg_chunks = (lsm_tree->merge_min + lsm_tree->merge_max) / 2;
+	for (last_size = avg_chunks * lsm_tree->chunk_size;
+	    last_size < chunk->size;
+	    last_size *= avg_chunks, ++chunk->generation) {}
+
```

Comment:
The generation that you are calculating here looks right to me.

I think this code might be a little clearer written like this:

```
avg_chunks = (lsm_tree->merge_min + lsm_tree->merge_max) / 2;
for (total_chunks = chunk->size / lsm_tree->chunk_size;
    total_chunks > 1;
    total_chunks /= avg_chunks)
        ++chunk->generation;
```